KIE forms jBPM Integration

pefernan opened the pull request:
- Added Providers for forms.

pefernan commented:
@mswiderski can you take a look?

mswiderski commented on kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/form/RemoteKieFormsProvider.java:
```diff
@@ -0,0 +1,140 @@
+import java.util.Iterator;
+import java.util.Map;
+import java.util.stream.Collectors;
+import javax.enterprise.context.Dependent;
```
could we avoid CDI annotations in kie server? and I guess Dependent is default if not defined, right?

mswiderski commented on kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/form/RemoteFormModellerFormProvider.java:
```diff
@@ -43,7 +43,7 @@
 import org.w3c.dom.NodeList;
 
-public class InMemoryFormProvider extends AbstractFormProvider implements UIFormProvider {
+public class RemoteFormModellerFormProvider extends AbstractFormProvider implements UIFormProvider {
```
what's the point of renaming class here?

mswiderski commented on kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/form/RemoteClassPathFormProvider.java:
```diff
@@ -0,0 +1,13 @@
+import org.jbpm.kie.services.impl.form.provider.InMemoryFormProvider;
+import org.kie.server.services.jbpm.ui.api.UIFormProvider;
+
+public class RemoteClassPathFormProvider extends InMemoryFormProvider implements UIFormProvider {
```
isn't the InMemoryFormProvider class was renamed?

mswiderski commented on kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/resources/META-INF/services/org.kie.server.services.jbpm.ui.api.UIFormProvider:
```diff
@@ -1 +1,4 @@
-org.kie.server.services.jbpm.ui.form.InMemoryFormProvider
\ No newline at end of file
+org.kie.server.services.jbpm.ui.form.RemoteFormModellerFormProvider
```
why do we have so many providers?

pefernan commented:
@mswiderski I added new providers to handle .ftl template forms. That's why I also renamed the existing provider, to match the existing provider names on jBPM with the ones here.

BTW, tests & licenses are comming, sorry for that.

pefernan commented:
@mswiderski added missing license headers & added all formtypes to test that the providing engine covers all form types.

mswiderski commented on kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/form/RemoteClassPathFormProvider.java:
```diff
@@ -0,0 +1,32 @@
+import org.jbpm.kie.services.impl.form.provider.InMemoryFormProvider;
+import org.kie.server.services.jbpm.ui.api.UIFormProvider;
+
+@Dependent
```
can we remove this?

mswiderski commented on kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/form/RemoteInMemoryFormProvider.java:
```diff
@@ -0,0 +1,32 @@
+import org.jbpm.kie.services.impl.form.provider.ClasspathFormProvider;
+import org.kie.server.services.jbpm.ui.api.UIFormProvider;
+
+@Dependent
```
same here, can we remove the @Dependent annotation?

mswiderski commented on kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/form/RemoteKieFormsProvider.java:
```diff
@@ -0,0 +1,156 @@
+import org.kie.api.task.model.Task;
+import org.kie.server.services.jbpm.ui.api.UIFormProvider;
+
+@Dependent
```
and here as well, can we remove it?

mswiderski commented:
@pefernan so we don't support .form (old form modeler format) anymore, right?

pefernan commented:
@mswiderski we do still support the old .form format. I haven't removed the old form engine yet, we still must decide what to do about it.

BTW: removed annotations.

pefernan commented:
jenkins retest this.

pefernan commented:
jenkins retest this.

pefernan commented:
@mswiderski can you take a look another time please??

mswiderski commented:
jenkins retest this

mswiderski commented:
changes look good, nice work! let's wait for jenkins to rerun the build.

mswiderski commented on kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/resources/META-INF/services/org.kie.server.services.jbpm.ui.api.UIFormProvider:
```diff
@@ -1 +1,4 @@
+org.kie.server.services.jbpm.ui.form.RemoteFormModellerFormProvider
+org.kie.server.services.jbpm.ui.form.RemoteKieFormsProvider
+org.kie.server.services.jbpm.ui.form.RemoteInMemoryFormProvider
+org.kie.server.services.jbpm.ui.form.RemoteClassPathFormProvider
```
RemoteClassPathFormProvider does not exists any more and that's why build is failing:

Caused by: java.util.ServiceConfigurationError: org.kie.server.services.jbpm.ui.api.UIFormProvider: Provider org.kie.server.services.jbpm.ui.form.RemoteClassPathFormProvider not found

pefernan commented:
jenkins retest this.

pefernan commented:
jenkins retest this.

pefernan commented:
jenkins retest this
